besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -258,5 +263,13 @@ private Long readQbftStartBlockConfig(final QbftConfigOptions qbftConfigOptions)
 
       return startBlock;
     }
+
+    private boolean isCheckpointBlockTotalDifficultyGreaterThanTTD(
+        final GenesisConfigOptions configOptions) {
+      return configOptions.getTerminalTotalDifficulty().isPresent()
+          && configOptions.getCheckpointOptions().isValid()
+          && (UInt256.fromHexString(configOptions.getCheckpointOptions().getTotalDifficulty().get())
+              .greaterThan(configOptions.getTerminalTotalDifficulty().get()));
```

Comment:
What happens if we're using a devnet or a new public testnet that starts with TTD = 0? Every blocks TD will be 0 and this condition will never be met?